src/components/action-popover/action-popover-menu.component.js

```diff
@@ -187,6 +199,22 @@ const ActionPopoverMenu = React.forwardRef(
   }
 );
 
+function checkRef(ref) {
+  return Boolean(ref && ref.current);
+}
+
+function getButtonHeight(buttonRef) {
+  // if no ref return height of default button
+  if (!checkRef(buttonRef)) {
```

Comment:
I think we need to look for a long term solution, we have a lot of similar code.

Something like this https://reach.tech/rect/ what do you think?

Reply:
Yeah I think something like this would definitely be worth it as we have a number of examples where we've had to default when the ref is null on first render